tests/SwaggerProvider.ProviderTests/Swagger.PetStore.Tests.fs

```diff
@@ -29,26 +28,82 @@ let petStoreTests =
         Expect.equal pet.Name "bar" "access modified value"
         Expect.stringContains (pet.ToString()) "bar" "ToString"
 
-    testCase "call provided methods" <| fun _ ->
+    testCaseAsync "call provided methods" <| async {
         try
-            store.DeletePet(1337L, apiKey)
+            do! store.DeletePet(1337L, Some apiKey)
         with
-        | exn -> ()
+        | _ -> ()
 
-        let tag = PetStore.Tag (Name = "foobar")
+        let tag = PetStore.Tag(None, Some "foobar")
         Expect.stringContains (tag.ToString()) "foobar" "ToString"
-        let pet = PetStore.Pet (Name = "foo", Id = Some 1337L, Status = "available")
+        let pet = PetStore.Pet("foo", [||], Some 1337L)
         Expect.stringContains (pet.ToString()) "1337" "ToString"
 
         try
-            store.AddPet(pet)
+            do! store.AddPet(pet)
         with
-        | exn -> failwithf "Adding pet failed with message: %s" exn.Message
+        | exn ->
+            let msg = if exn.InnerException = null then exn.Message
+                      else exn.InnerException.Message
+            failwithf "Adding pet failed with message: %s" msg
 
-        let pet2 = store.GetPetById(1337L)
+        let! pet2 = store.GetPetById(1337L)
         Expect.equal pet.Name     pet2.Name     "same Name"
         Expect.equal pet.Id       pet.Id        "same Id"
         Expect.equal pet.Category pet2.Category "same Category"
         Expect.equal pet.Status   pet2.Status   "same Status"
         Expect.notEqual pet pet2 "different objects"
+    }
+
+    testCase "create types with Nullable properties" <| fun _ ->
+        let tag = PetStoreNullable.Tag(Nullable<_>(), "foobar")
+        Expect.stringContains (tag.ToString()) "foobar" "ToString"
+        let tag2 = PetStoreNullable.Tag (Name = "foobar")
+        Expect.stringContains (tag2.ToString()) "foobar" "ToString"
+
+        let pet = PetStoreNullable.Pet("foo", [||], Nullable(1337L))
+        Expect.stringContains (pet.ToString()) "1337" "ToString"
+        let pet2 = PetStoreNullable.Pet (Name="foo", Id = Nullable(1337L))
+        Expect.stringContains (pet2.ToString()) "1337" "ToString"
+  ]
+
+open System.Net
+open System.Net.Http
+open System.Threading.Tasks
+
+type PetStoreServer = SwaggerApiProvider<"http://petstore.swagger.io/v2/swagger.json">
+let server =
+    PetStoreServer.Handler(
+        AddPet = fun req ->
+            let response = new HttpResponseMessage(Net.HttpStatusCode.Created, RequestMessage = req)
+            Task.FromResult(response)
+    )
```

Comment:
Assuming `req` and `response` become generated types, what do you think of having the operation handlers be property setters to avoid overwriting already defined handlers?